src/Components/Nav/Nav.jsx

```diff
@@ -1,43 +1,86 @@
 import React, { Component } from "react";
-import { Link } from "react-router-dom";
-import styled from "styled-components";
+import { NavLink, withRouter } from "react-router-dom";
+import styled, { css } from "styled-components";
+
 class Nav extends Component {
   render() {
+    const { transparent, history } = this.props;
+    const isAuth = !!localStorage.getItem("token");
+    const loginUser = () => {
+      localStorage.setItem("token");
+      history.push("/");
+    };
+
+    const logoutUser = () => {
+      localStorage.removeItem("token");
+      history.push("/");
+    };
     return (
-      <NavBar>
+      <NavBar transparent={transparent}>
         <CenterWrapper width="1060px" className="global">
           <Wrapper>
             <Logo>
-              <Link to="/">
+              <NavLink
+                to="/"
+                onClick={() => {
+                  window.location.href = "/";
+                }}
+              >
                 <img alt="main_logo" src="./images/main_logo.svg" />
-              </Link>
+              </NavLink>
             </Logo>
-            <SearchBar>
+            <SearchBar transparent={transparent}>
               <i class="fas fa-search"></i>
               <input
                 type="search"
                 placeholder="도시나 상품을 검색해보세요"
               ></input>
             </SearchBar>
           </Wrapper>
-          <List className="global">
-            <li>위시리스트</li>
-            <li>예약내역</li>
-            <li>메시지</li>
-            <li>알림</li>
+          <List className="global" transparent={transparent}>
+            <li>파트너 등록하기</li>
+            <NavLink
+              to="/login"
+              onClick={() => {
+                window.location.href = "/login";
+              }}
+            >
+              {!isAuth ? (
+                <li onClick={loginUser}>로그인</li>
+              ) : (
+                <li onClick={logoutUser}>로그아웃</li>
+              )}
```

Comment:
삼항연산자로 잘 구현해주셨습니다! 아래의 사항만 수정해주시면 더 좋은 코드가 될 수 잇겠네요

- 한 조건에만 벌써 부정이 세번이 들어갔네요! 위 조건문에서도 부정을 빼주시고 조건에 따라 실행문의 순서만 달리 할 수 있겠습니다.
- 실행문도 비슷한 코드가 반복되는 것처럼 보이네요. onClick이벤트에서 실행되는 함수를 하나의 함수로 바꾸면 아래와 같이 좀 더 간결하게 작성 할 수 있습니다.
```suggestion
              <li onClick={handleLoginState}>{isAuth ? "로그아웃" : "로그인"}</li>
```